### DISCUSSION TITLE
Allow to edit embed link (like draw.io)

### DISCUSSION THREAD

**Discussion by max13fr** [outline/outline#6973] [Ideas]
Hello,

It would be awesome to allow editing the link of the embed components like draw.io / figma / canvas / ...

Currently, we need to drop the embed and recreate a new one with the modified link.

Thanks in advance,
Best regards, Max

**Comment by notfounnd** [comment]
I was going to open exactly an idea like this for Outline.

Keeping diagrams in the Outline tool is quite laborious in the way the design of the functionality is done.

Maybe initially it doesn’t need to be for all tools. I suggest starting with draw.io because I believe it meets most of the needs.

I believe that if this option of editing in an embedded way was possible for graphics made in draw.io we would already have a very large value being added to the tool.

This suggested model of use with draw.io can be experienced in WikiJS 2.0.

- https://www.youtube.com/watch?v=xjOfL_mMWk4&t=717s (diagrams feature example video)
- https://www.drawio.com/integrations
- https://github.com/requarks/wiki/blob/main/server/modules/rendering/html-diagram/renderer.js
- https://github.com/requarks/wiki/blob/main/client/components/editor/editor-modal-drawio.vue

In the text editing screen, when adding a draw.io chart you are directed to the drawing tool editor. When you finish editing the chart, draw.io puts you back on the page that was editing and renders the drawing done.

If you want to edit an existing chart, just click on the edit button and it opens draw.io with the current drawing.

**Comment by rockmeira** [comment]
That's a great idea! Allowing the editing of links in embedded components would definitely make working with tools like [draw.io](http://draw.io/) and other similar ones much more practical. The suggestion to start with [draw.io](http://draw.io/) makes perfect sense, as it's such a popular tool.

**Comment by rafaelbandeeira** [comment]
Oh I would love this feature idea! It would make the experience of working with those tools much more seamless, and fluid.

**Comment by Tiago-Verde** [comment]
It's a wonderful idea and this can make our experience much more easy and productive.

**Comment by notfounnd** [comment]
I think it was resolved in [v1.2.0](https://github.com/outline/outline/releases/tag/v1.2.0).

**Comment by max13fr** [comment]
I confirm, thanks you very much to the devs implied !
